index.d.ts

```diff
@@ -1,52 +1,248 @@
-export type ElementId = string;
+/**
+ * Represents a rendered Turnstile widget.  Used to identify a specific widget when calling
+ * Turnstile methods.
+ */
 export type WidgetId = string;
-export type Container = ElementId | HTMLElement;
 
-declare global {
-	interface Window {
-		turnstile: TurnstileObject;
-	}
-}
+/**
+ * An HTML element id used as a selector.
+ */
+export type ElementId = string;
 
+/**
+ * Turnstile is Cloudflare’s smart CAPTCHA alternative. It can be embedded into any website
+ * without sending traffic through Cloudflare and works without showing visitors a CAPTCHA.
+ * It's accessible via `window.turnstile`.
+ * @see https://developers.cloudflare.com/turnstile
+ */
 export interface TurnstileObject {
+	/**
+	 * If using synchronous loading, will be called once the DOM is ready.
+	 * @see [Explicit Rendering](https://developers.cloudflare.com/turnstile/get-started/client-side-rendering/#explicitly-render-the-turnstile-widget)
+	 */
 	ready: (cb: () => any) => void;
+
+	/**
+	 * @see [Implicit Rendering](https://developers.cloudflare.com/turnstile/get-started/client-side-rendering/#implicitly-render-the-turnstile-widget)
+	 */
 	implicitRender: () => void;
-	execute: (container?: Container, options?: TurnstileOptions) => void;
-	render: (container: Container, options: TurnstileOptions) => WidgetId;
-	reset: (widget?: Container | WidgetId) => void;
-	remove: (widget?: Container | WidgetId) => void;
-	getResponse: (widget?: Container | WidgetId) => string | undefined;
-	isExpired: (widget?: Container | WidgetId) => boolean;
+
+	/**
+	 * Excecutes the challenge after the render() function has been called, by invoking the
+	 * turnstile.execute(container: string | HTMLElement, jsParams?: RenderParameters) function
+	 * separately, decoupling the appearance and rendering of a widget from its execution.
+	 * @param container - HTML container to render the widget, or an element id selector.
+	 * @param options - Configuration options for rendering. See {@link RenderParameters}.
+	 * @see [Execution Modes](https://developers.cloudflare.com/turnstile/get-started/client-side-rendering/#execution-modes)
+	 */
+	execute: (container?: HTMLElement | ElementId, options?: TurnstileOptions) => void;
+
+	/**
+	 * Explicitly renders the Turnstile widget.  The `sitekey` parameter is required. The
+	 * `callback` parameter, however, is only used if `execution` is set to `execute`.
+	 * @param container - HTML container to render the widget, or an element id selector.
+	 * @param options - Configuration options for rendering. See {@link RenderParameters}.
+	 * @returns A `widgetId` string if successful, otherwise undefined.
+	 */
+	render(container: HTMLElement | ElementId, options: RenderParameters): string | undefined;
```

Comment:
```suggestion
	 * @param parameters - Configuration options for rendering. See {@link RenderParameters}.
	 * @returns A `widgetId` string if successful, otherwise undefined.
	 */
	render(container: HTMLElement | ElementId, parameters: RenderParameters): string | undefined;
```